arrow-libs/optics/arrow-optics/src/commonMain/kotlin/arrow/optics/Fold.kt

```diff
@@ -25,34 +26,41 @@ import kotlin.jvm.JvmStatic
  */
 public interface Fold<S, A> {
 
+  /**
+   * Map each target to a type [R] and combine the results as a fold.
+   */
+  public fun <R> foldMap(empty: R, combine: (R, R) -> R, source: S, map: (focus: A) -> R): R
+
   /**
    * Map each target to a type R and use a Monoid to fold the results
    */
-  public fun <R> foldMap(M: Monoid<R>, source: S, map: (focus: A) -> R): R
+  @Deprecated(MonoidDeprecation, ReplaceWith("foldMap(M.empty(), M::combine, source, map)", "arrow.typeclasses.combine"))
+  public fun <R> foldMap(M: Monoid<R>, source: S, map: (focus: A) -> R): R =
+    foldMap(M.empty(), M::combine, source, map)
```

Comment:
I think the only way to do this in a non-binary breaking change is to:

 - Deprecate `foldMap` with `Monoid` but keep it abstract
 - Implement the new `foldMap` through the abstract method. (This would require creating an anonymous `Monoid`)

Reply:
This would meant that someone who implements a custom `Fold` needs to refactor a breaking change in `2.0`, but a consumer would not face a breaking change when not relying on the deprecated code.

Reply:
I think I've managed to keep ABI compatibility by:
- Keep the `Fold` interface as it was, using `Monoid`;
- Define the new `fold` and `foldMap` as extension functions, so they don't become part of the interface.